starknet_py/net/account/account.py

```diff
@@ -0,0 +1,394 @@
+    ) -> int:
+        """
+        Checks account's balance of specified token.
+
+        :param token_address: Address of the ERC20 contract.
+            If not specified it will be payment token (wrapped ETH) address.
+        :return: Token balance
+        """
+
+        token_address = token_address or self._get_default_token_address()
+
+        low, high = await self._client.call_contract(
+            Call(
+                to_addr=parse_address(token_address),
+                selector=get_selector_from_name("balanceOf"),
+                calldata=[self.address],
+            ),
+            block_hash="pending",
+        )
+
+        return (high << 128) + low
+
+    async def sign_invoke_transaction(
+        self,
+        calls: Calls,
+        *,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> InvokeFunction:
+        """
+        Takes calls and creates signed InvokeFunction
+
+        :param calls: Single call or list of calls
+        :param max_fee: Max amount of Wei to be paid when executing transaction
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
+        :return: InvokeFunction created from the calls
+        """
+
+        execute_tx = await self._prepare_invoke_function(calls, max_fee, auto_estimate)
+        signature = self.signer.sign_transaction(execute_tx)
+        return _add_signature_to_transaction(execute_tx, signature)
+
+    async def sign_declare_transaction(
+        self,
+        compiled_contract: str,
+        *,
+        cairo_path: Optional[List[str]] = None,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> Declare:
+        """
+        Create and sign declare transaction.
+
+        :param compiled_contract: string containing compiled contract bytecode.
+                                  Useful for reading compiled contract from a file
+        :param cairo_path: a ``list`` of paths used by starknet_compile to resolve dependencies within contracts
+        :param max_fee: Max amount of Wei to be paid when executing transaction
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
+        :return: Signed Declare transaction
+        """
+        # pylint: disable=too-many-arguments
+        compiled_contract = create_compiled_contract(
+            compiled_contract=compiled_contract, search_paths=cairo_path
+        )
+        declare_tx = Declare(
+            contract_class=compiled_contract,
+            sender_address=self.address,
+            max_fee=0,
+            signature=[],
+            nonce=await self.get_nonce(),
+            version=self._version,
+        )
+
+        max_fee = await self._get_max_fee(
+            transaction=declare_tx, max_fee=max_fee, auto_estimate=auto_estimate
+        )
+        declare_tx = dataclasses.replace(declare_tx, max_fee=max_fee)
```

Comment:
Can be replaced with `_add_max_fee_to_transaction` like the signature was.
We use it in the `_prepare_invoke_transaction` too

Reply:
Or maybe we could create a function which changes any field we want? :hmm:

Reply:
You mean like `dataclasses.replace` does exactly? But I guess we can abstract it to `_add_max_fee_to_transaction`, keep the steps more obvious.